### PR TITLE
DLS-9129: Remove keystore from CBC frontend

### PR DESCRIPTION
**Pull request by johno1985**

### CODE REVIEW

**Comment by johno1985** [comment]
rebuild

File: app/uk/gov/hmrc/cbcrfrontend/services/ReportingEntityDataService.scala [hunk 1]
```diff
@@ -55,12 +55,13 @@ class ReportingEntityDataService @Inject()(connector: CBCRBackendConnector)(impl
                 .validate[ReportingEntityData]
                 .fold(
                   failed =>
-                    Left(UnexpectedState(s"Unable to serialise response as ReportingEntityData: ${failed.mkString}")),
+                    Left(UnexpectedState(s"Unable to deserialise response as ReportingEntityData: ${failed.mkString}")),
```
Comment (LAJW) [inline app/uk/gov/hmrc/cbcrfrontend/services/ReportingEntityDataService.scala:58]: ⛏️ Not quite a compiler warning, rather fixing error messages.

File: test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala [hunk 2]
```diff
@@ -446,8 +443,8 @@ class FileUploadControllerSpec
         val enrolment = CBCEnrolment(cbcId, Utr("7000000002"))
         fuService.getFile(*, *) returnsF validFile
         fuService.getFileMetaData(*, *)(*) returnsF Some(md.copy(name = "bad.zip"))
-        cache.read[CBCId](CBCId.cbcIdFormat, *, *) returnsF CBCId.create(1).getOrElse(fail("baaa"))
-        cache.save(*)(*, *, *) returns Future.successful(CacheItem("", JsObject.empty, Instant.now(), Instant.now()))
+        cache.save[FileMetadata](*)(FileMetadata.fileMetadataFormat, *, *) returns Future.successful(
+          CacheItem("", JsObject.empty, Instant.now(), Instant.now()))
```
Comment (LAJW) [inline test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala:447]: ❓ Why has this test changed?

File: app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala [hunk 3]
```diff
@@ -18,66 +18,63 @@ package uk.gov.hmrc.cbcrfrontend.services
...
+    extends MongoCacheRepository(
+      mongoComponent = mongoComponent,
+      collectionName = "session-data",
+      ttl = config.get[FiniteDuration]("mongodb.session.expireAfter"),
+      timestampSupport = timestampSupport,
+      cacheIdType = CacheIdType.SimpleCacheId
+    ) with Logging {
 
-  private val conf = config.underlying.get[Config]("microservice.services.cachable.session-cache").value
+  private val noSession = Future.failed[String](NoSessionException)
```
[the remark this answers is not in the document]
Comment (johno1985) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala:47]: Why indeed, updating PR....

File: app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala [hunk 4]
```diff
@@ -18,66 +18,63 @@ package uk.gov.hmrc.cbcrfrontend.services
...
-        }
-      }
-      .recover {
-        case NonFatal(t) =>
-          logger.info(s"CBCSessionCache Failed - error message: ${t.getMessage}")
-          false
-      }
+    for {
+      cacheId <- extractCacheId
+      result  <- deleteEntity(cacheId).map(_ => true)
```
[the remark this answers is not in the document]
Comment (johno1985) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala:78]: Not sure where else to put it?

File: app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala [hunk 5]
```diff
@@ -18,66 +18,63 @@ package uk.gov.hmrc.cbcrfrontend.services
...
+      cacheId <- extractCacheId
+      result  <- put[T](cacheId)(DataKey[T](extractType), body)
+    } yield result
 
   def read[T: Reads: TypeTag](implicit hc: HeaderCarrier): EitherT[Future, ExpiredSession, T] =
     EitherT[Future, ExpiredSession, T](
       readOption[T]
-        .map(_.toRight(ExpiredSession(s"Unable to read ${typeOf[T]} from cache")))
-    )
+        .map(_.toRight(ExpiredSession(s"Unable to read $extractType from cache"))))
```
[the remark this answers is not in the document]
Comment (johno1985) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala]: Updating...

File: test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala (hunk 2 above)
Comment (johno1985) [reply test/uk/gov/hmrc/cbcrfrontend/controllers/FileUploadControllerSpec.scala:447]: There was no cache read taking place, so I removed the mock entry. I updated the cache.save mock statement, passing in the proper type arguments

File: test/uk/gov/hmrc/cbcrfrontend/controllers/SharedControllerSpec.scala [hunk 6]
```diff
@@ -307,8 +312,9 @@ class SharedControllerSpec
       cache.readOption[Utr](Utr.utrRead, *, *) returns Future.successful(None)
       cache.readOption[CompleteXMLInfo](CompleteXMLInfo.format, *, *) returns Future.successful(None)
       cache.save[BusinessPartnerRecord](*)(*, *, *) returns Future.successful(
-        CacheMap("cache", Map.empty[String, JsValue]))
-      cache.save[Utr](*)(*, *, *) returns Future.successful(CacheMap("cache", Map.empty[String, JsValue]))
+        CacheItem("id", JsObject.empty, Instant.now(), Instant.now()))
+      cache.save[Utr](*)(*, *, *) returns Future.successful(
+        CacheItem("id", JsObject.empty, Instant.now(), Instant.now()))
```
[the remark this answers is not in the document]
Comment (johno1985) [reply test/uk/gov/hmrc/cbcrfrontend/controllers/SharedControllerSpec.scala:323]: Updating...

File: app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala (hunk 4 above)
[the remark this answers is not in the document]
Comment (LAJW) [reply app/uk/gov/hmrc/cbcrfrontend/services/CBCSessionCache.scala:78]: When for is in use, then we don't need maps.
```scala
_ <- deleteEntity(cacheId)
yield true
```

**Review by LAJW — DISMISSED** [review]
Question answered, approving.

**Review by LAJW — APPROVED** [review]

**Pull request merged**